clinica/iotools/bids_utils.py

```diff
@@ -523,168 +534,247 @@ def write_modality_agnostic_files(
 
 
 def get_bids_subjs_list(bids_path: str) -> List[str]:
-    """Given a BIDS compliant dataset, return the list of all the subjects available."""
+    """Given a BIDS compliant dataset, return the list of all the subjects available.
+
+    Parameters
+    ----------
+    bids_path : str
+        Path to the BIDS directory.
+
+    Returns
+    -------
+    List[str] :
+        List of subject IDs available in this BIDS dataset.
+
+    See also
+    --------
+    get_bids_sess_list
+    get_bids_subjs_paths
+    """
     from pathlib import Path
 
     return [str(d.name) for d in Path(bids_path).glob("sub-*") if d.is_dir()]
 
 
 def get_bids_sess_list(subj_path: str) -> List[str]:
-    """
-    Given a subject path, return the list of sessions available
+    """Given a path to a subject's folder, this function returns the
+    list of sessions available.
+
+    Parameters
+    ----------
+    subj_path : str
+        Path to the subject folder for which to list the sessions.
+
+    Returns
+    -------
+    List[str] :
+        The list of session names for this subject.
+
+    See also
+    --------
+    get_bids_subjs_list
+    get_bids_subjs_paths
     """
     from pathlib import Path
 
     return [str(d.name) for d in Path(subj_path).glob("ses-*") if d.is_dir()]
 
 
 def get_bids_subjs_paths(bids_path: str) -> List[str]:
-    """Given a BIDS compliant dataset, returns the list of all paths to the subjects folders."""
+    """Given a BIDS compliant dataset, returns the list of all paths to the subjects folders.
+
+    Parameters
+    ----------
+    bids_path : str
+        Path to the BIDS directory.
+
+    Returns
+    -------
+    List[str] :
+        List of paths to the subjects folders.
+
+    See also
+    --------
+    get_bids_subjs_list
+    get_bids_sess_list
+    """
     from pathlib import Path
 
     return [str(d) for d in Path(bids_path).glob("sub-*") if d.is_dir()]
 
 
-def remove_space_and_symbols(data):
-    """Remove spaces and  - _ from a list (or a single) of strings.
+def remove_space_and_symbols(data: Union[str, List[str]]) -> Union[str, List[str]]:
+    """Remove spaces, "-", and "_" characters from a string.
 
-    Args:
-        data: list of strings or a single string to clean
+    If a list of strings is provided, this function will be called
+    on each item of the list.
 
-    Returns:
-        data: list of strings or a string without space and symbols _ and -
+    Parameters
+    ----------
+    data : str or List[str]
+        String or list of strings to be cleaned.
+
+    Returns
+    -------
+    str or List[str] :
+        Cleaned string(s).
     """
     import re
 
-    if type(data) is list:
-        for i in range(0, len(data)):
-            data[i] = re.sub("[-_ ]", "", data[i])
-    else:
-        data = re.sub("[-_ ]", "", data)
+    if isinstance(data, list):
+        return [remove_space_and_symbols(d) for d in data]
+    return re.sub("[-_ ]", "", data)
```

Comment:
Using a more Pythonic pattern:
```suggestion
    try:
        return [remove_space_and_symbols(d) for d in data]
    except TypeError:
        return re.sub("[-_ ]", "", data)
```

Also allow data to be `str | Iterable[str]` instead of just `str | list[str]`.

Reply:
Hmmm, I accepted the suggestion too fast. I really need to check if `data` is a list since the function is recursive.